test/hotel_booker_test.rb

```diff
@@ -0,0 +1,101 @@
+require_relative "test_helper"
+
+describe "Wave 1 " do
+  describe "initialize" do
+    before do
+      @hotel = HotelBooker.new
+    end
+    it "creates  20 rooms " do
+      expect(@hotel.rooms.length).must_equal 20
+    end
+    it "checks that rooms are integers" do
+      expect(@hotel.rooms).must_be_instance_of Array
+      @hotel.rooms.each do |room|
+        expect(room).must_be_instance_of Integer
+      end
+    end
+  end
+
+  describe " create_reservation" do
+    it "returns a reservation" do
+      hotel = HotelBooker.new
+      reservation = hotel.create_reservation(1, Date.new(2019, 2, 3), Date.new(2019, 2, 6))
+      expect(reservation).must_be_instance_of Reservation
+    end
+
+    it "checks if new reservation is added to reservations" do
+      hotel = HotelBooker.new
+      hotel.create_reservation(1, Date.new(2019, 2, 5), Date.new(2019, 2, 9))
+      expect(hotel.reservations[1].length).must_equal 1
+    end
+    it "doesnt make conflicting reservations for the same room" do
+      hotel = HotelBooker.new
+      hotel.create_reservation(1, Date.new(2019, 2, 6), Date.new(2019, 2, 11))
+      hotel.create_reservation(4, Date.new(2019, 2, 7), Date.new(2019, 2, 10))
+
+      expect(hotel.reservations[1].length).must_equal 1
+      expect(hotel.reservations[2].length).must_equal 1
+    end
+    it "checks that new reservations can start on checkout days" do
```

Comment:
I wanna point out that there's some inconsistency around naming. Some of your test names describe what the source code should do and some describe what the test does. Conventionally, the former is preferred.